### PR TITLE
Add min max normalization diffusion policy

### PR DESCRIPTION
**Pull request by kwangneuraco**
### Features

- Add min_max normalization in diffusion policy
- Modified several hyper-parameters to get better performance
- Modified general config.yaml to match training config in frontend and our default training setting up.
- Add output_types check for normalization in other methods

Related to ticket
[NCRL-47](https://neuracore.atlassian.net/browse/NCRL-47?atlOrigin=eyJpIjoiMGM5OGZiMDM3ZWQxNGY5ZGFhNzA1NGFmMTUxMjdlNmMiLCJwIjoiaiJ9)

### CODE REVIEW

File: neuracore/ml/algorithms/diffusion_policy/diffusion_policy.py [hunk 1]
```diff
@@ -183,17 +237,67 @@ def _to_torch_float_tensor(self, data: list[float]) -> torch.FloatTensor:
...
+            range_val = stat2 - stat1
+            # Avoid division by zero
+            range_val = torch.clamp(range_val, min=1e-8)
+            # Scale to [0, 1], then map to [-1, 1]
+            normalized = (joint_state - stat1) / range_val
+            return normalized * 2.0 - 1.0
+        else:
+            raise ValueError(
+                f"Unsupported normalization_type: {self.normalization_type}"
+            )
```
Comment (aditya-neuraco) [inline neuracore/ml/algorithms/diffusion_policy/diffusion_policy.py:277]: We can pass the `dataset_description` directly and read stats from there. The code would be cleaner.

File: neuracore/ml/algorithms/pi0/pi0.py [hunk 2]
```diff
@@ -215,18 +215,26 @@ def _setup_normalization_stats(self) -> None:
...
+            )
+            self.register_buffer(
+                "joint_target_std",
+                self._to_torch_float_tensor(
+                    self.dataset_description.joint_target_positions.std
+                ),
+            )
+        else:
+            self.joint_target_mean = None
+            self.joint_target_std = None
```
Comment (aditya-neuraco) [inline neuracore/ml/algorithms/pi0/pi0.py]: This could be in a different PR? It's changing files in `Pi0` but this PR is seems mainly about Diffusion Policy.

File: neuracore/ml/algorithms/simple_vla/simple_vla.py [hunk 3]
```diff
@@ -162,18 +162,26 @@ def _setup_normalization_stats(self) -> None:
...
+            )
+            self.register_buffer(
+                "joint_target_std",
+                self._to_torch_float_tensor(
+                    self.dataset_description.joint_target_positions.std
+                ),
+            )
+        else:
+            self.joint_target_mean = None
+            self.joint_target_std = None
```
Comment (aditya-neuraco) [inline neuracore/ml/algorithms/simple_vla/simple_vla.py]: This could be in a different PR? It's changing files in `SimpleVLA` but this PR is seems mainly about Diffusion Policy.

File: neuracore/ml/config/algorithm/diffusion_policy.yaml [hunk 4]
```diff
@@ -2,20 +2,22 @@
...
   beta_end: 0.02
   beta_schedule: "squaredcos_cap_v2"
   clip_sample: true
   clip_sample_range: 1.0
   lr: 1e-4
   lr_backbone: 1e-5
-  weight_decay: 1e-4
+  weight_decay: 1e-6
   prediction_type: "epsilon"
+  normalization_type: "min_max"  # Options: "mean_std" or "min_max"
```
Comment (aditya-neuraco) [inline neuracore/ml/config/algorithm/diffusion_policy.yaml]: Don't need a comment here as we are not commenting the choices for other options.

File: neuracore/ml/config/config.yaml [hunk 5]
```diff
@@ -5,8 +5,8 @@ defaults:
 
 # Core training parameters
 seed: 42
-epochs: 100
-output_prediction_horizon: 10
+epochs: 10
+output_prediction_horizon: 100
```
Comment (aditya-neuraco) [inline neuracore/ml/config/config.yaml]: I think we should keep the default number of epochs as `100` because `10` epochs won't lead to acceptable performance on any dataset/algorithm.

File: neuracore/ml/algorithms/pi0/pi0.py (hunk 2 above)
Comment (kwangneuraco) [reply neuracore/ml/algorithms/pi0/pi0.py]: ok sure

File: neuracore/ml/algorithms/diffusion_policy/diffusion_policy.py (hunk 1 above)
Comment (kwangneuraco) [reply neuracore/ml/algorithms/diffusion_policy/diffusion_policy.py:277]: Agree, we should have a dataset_stats dictionary with all the elements (mean, std, etc), and setup_normalization function just returns this. I would rather put this into a separate PR in which the normalization and un-normalization are properly handled and as a helper function.
